pantheon/src/main/java/tech/pegasys/pantheon/controller/MainnetPantheonController.java

```diff
@@ -116,6 +123,9 @@ public MainnetPantheonController(
             metricsSystem);
     final SyncState syncState =
         new SyncState(blockchain, ethProtocolManager.ethContext().getEthPeers());
+    final LabelledMetric<OperationTimer> ethTasksTimer =
+        metricsSystem.createLabelledTimer(
+            MetricCategory.SYNCHRONIZER, "task", "Internal processing tasks", "taskName");
```

Comment:
What was the reason for pushing this up to here?  Seems odd to have only done it for `MainnetPantheonController` and not clique & IBFT, plus we're generally trying to push common construction code out of the controller.

Reply:
Because we need it to construct any `EthTask`'s and `DaoForkPeerValidator` uses an `EthTask`.  I think the real fix is to keep working on `MetricsSystem` to allow metrics to be created idempotently so we can then push this timer creation into `AbstractEthTask` where it's used.  I have some in-progress work on that which I can come back to.

Reply:
ahhh, yeah that's very unfortunate.

Reply:
Just remembered, I added some caching for metrics back when I did the EVM metrics but had to revert it, so https://github.com/PegaSysEng/pantheon/commit/525aa19d5075fa376341715c19a15152e1554642#diff-b95472676d731050caa643c3ebe592b1 may be useful.